vivisect/tests/testelf.py

```diff
@@ -63,82 +63,228 @@ def setUpClass(cls):
+                logger.warn("rel: o: %-80s\tn: %s" % (equiv, newname))
+            done.append(va)
+
+        return failed_old, failed_new
 
 
     def names(self, vw, data, fname):
-        # simple comparison to ensure same workspace names.  perhaps too simple.
-        newnames = vw.getNames()
+        # comparison to ensure same workspace names
+
+        # filter out a lot of noise not likely to be indicative of ELF bugs.
+        newnames = [ntup for ntup in vw.getNames() if not (
+                ntup[1].startswith('str_') or
+                ntup[1].startswith('ptr_str_') or
+                ntup[1].startswith('ptr_sub_') or
+                ntup[1].startswith('sub_'))]
+
         newnames.sort()
         oldnames = data['names']
         oldnames.sort()
-        self.assertListEqual(newnames, newnames)
+
+        failed_new = 0
+        failed_old = 0
+        done = []
         for oldname in oldnames:
-            if oldname[1].startswith('str_') or oldname[1].startswith('ptr_str_') \
-                    or oldname[1].startswith('ptr_sub_') or oldname[1].startswith('sub_'):
-                continue
-            self.assertIn(oldname, newnames, msg='names: missing: %r   (%r)' % (oldname, fname))
+            va = oldname[0]
+            equiv = None
+            done.append(va)
+
+            for newname in newnames:
+                if newname[0] == va:
+                    equiv = newname
+                    break
+            if oldname != equiv:
+                failed_old += 1
+                logger.warn("name: o: %-80s\tn: %s" % (oldname, equiv))
+
         for newname in newnames:
-            if newname[1].startswith('str_') or newname[1].startswith('ptr_str_') \
-                    or newname[1].startswith('ptr_sub_') or newname[1].startswith('sub_'):
+            va = newname[0]
+            if va in done:
                 continue
-            self.assertIn(newname, oldnames, msg='names: new: %r   (%r)' % (newname, fname))
+
+            equiv = None
+            done.append(va)
+
+            for oldname in oldnames:
+                if oldname[0] == va:
+                    equiv = oldname
+                    break
+            if newname != equiv:
+                failed_new += 1
+                logger.warn("name: o: %-80s\tn: %s" % (equiv, newname))
+
+        return failed_old, failed_new
 
     def pltgot(self, vw, data, fname):
         for pltva, gotva in data['pltgot']:
             match = False
             for xfr, xto, xtype, xinfo in vw.getXrefsFrom(pltva):
                 if xfr == pltva and xto == gotva:
                     match = True
-            self.assertEqual((hex(pltva), match), (hex(pltva), True), msg='pltgot: %r' % fname)
+
+        return 0,0
 
     def debuginfosyms(self, vw, data, fname):
         # we don't currently parse debugging symbols.
         # while they are seldom in hard targets, this is a weakness we should correct.
-        pass
+        return 0,0
```

Comment:
That reminds me. I've got a dwarfinfo branch out still. I should go back and update that.

Reply:
:+1: